### PR TITLE
WFLY-18179 & WFLY-18230 Avoid redundant no-op runtime steps in undertow and security subsystems

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-18179
https://issues.redhat.com/browse/WFLY-18230

... and assorted changes required by https://github.com/wildfly/wildfly-core/pull/5563

### CODE REVIEW

File: jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsSubsystemDefinition.java [hunk 1]
```diff
@@ -63,6 +64,13 @@ public class JaxrsSubsystemDefinition extends SimpleResourceDefinition {
                  .setRemoveHandler(ReloadRequiredRemoveStepHandler.INSTANCE));
     }
 
+    @Override
+    public void registerAttributes(ManagementResourceRegistration registration) {
+       for (AttributeDefinition definition : JaxrsAttribute.ATTRIBUTES) {
+          registration.registerReadWriteAttribute(definition, null, new JaxrsParamHandler(definition));
```
Comment (bstansberry) [inline jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsSubsystemDefinition.java:70]: FYI re https://issues.redhat.com/browse/WFLY-18257. I'm not requesting a change; just FYI.

File: undertow/src/main/java/org/wildfly/extension/undertow/filters/FilterDefinitions.java [hunk 2]
```diff
@@ -45,8 +45,8 @@ public class FilterDefinitions extends PersistentResourceDefinition {
 
     public FilterDefinitions() {
         super(PATH_ELEMENT, UndertowExtension.getResolver(PATH_ELEMENT.getValue()),
-                new AbstractAddStepHandler(),
-                ReloadRequiredRemoveStepHandler.INSTANCE
+                new ModelOnlyAddStepHandler(),
+                ModelOnlyRemoveStepHandler.INSTANCE
```
Comment (bstansberry) [inline undertow/src/main/java/org/wildfly/extension/undertow/filters/FilterDefinitions.java:49]: This remove handler change seems out of scope and perhaps changes the reported read-operation-description behavior. There were some recent bug-fixing changes in this area and I'm reluctant to change this as part of this work.

File: undertow/src/main/java/org/wildfly/extension/undertow/handlers/HandlerDefinitions.java [hunk 3]
```diff
@@ -43,8 +43,8 @@ public class HandlerDefinitions extends PersistentResourceDefinition {
 
     public HandlerDefinitions() {
         super(new SimpleResourceDefinition.Parameters(PATH_ELEMENT, UndertowExtension.getResolver(PATH_ELEMENT.getValue()))
-                .setAddHandler(new AbstractAddStepHandler())
-                .setRemoveHandler(ReloadRequiredRemoveStepHandler.INSTANCE)
+                .setAddHandler(new ModelOnlyAddStepHandler())
+                .setRemoveHandler(ModelOnlyRemoveStepHandler.INSTANCE)
```
Comment (bstansberry) [inline undertow/src/main/java/org/wildfly/extension/undertow/handlers/HandlerDefinitions.java:47]: Same comment as for the remove handler in FilterDefinitions.

**Review by bstansberry — COMMENTED** [review]
@pferraro This all looks fine, except I'm concerned about the 2 undertow remove handler changes.

File: undertow/src/main/java/org/wildfly/extension/undertow/filters/FilterDefinitions.java (hunk 2 above)
Comment (pferraro) [reply undertow/src/main/java/org/wildfly/extension/undertow/filters/FilterDefinitions.java:49]: This change is the essence of this PR. This resource has no runtime - thus the existing handlers create a completely unnecessary runtime step.
If this resource contains any child resources (i.e. actual filter resource with runtime handling), removing this resource will recursively remove its children and put the server into reload required.

File: jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsSubsystemDefinition.java (hunk 1 above)
Comment (pferraro) [reply jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsSubsystemDefinition.java:70]: That's worthwhile change to make now (though, eventually, these attribute definition parameters will be made obsolete by https://github.com/wildfly/wildfly-core/pull/5563).

File: undertow/src/main/java/org/wildfly/extension/undertow/filters/FilterDefinitions.java (hunk 2 above)
Comment (pferraro) [reply undertow/src/main/java/org/wildfly/extension/undertow/filters/FilterDefinitions.java:49]: > This remove handler change seems out of scope and perhaps changes the reported read-operation-description behavior.

Can you elaborate?  I don't see any recent changes to the ReadOperationDescriptionHandler.  Yes, this would change the reported "restart-required" value for the remove operation - but that's the whole point.

Comment (bstansberry) [reply undertow/src/main/java/org/wildfly/extension/undertow/filters/FilterDefinitions.java:49]: Ok; I see now both of these resources are direct children of the subsystem root and truly have no runtime behavior, i.e. no parent whose behavior is influenced by these.

I believe the thing I was talking about related to FilterRefDefinition, which is a different case.

Thanks, and sorry for the noise.

**Review by bstansberry — APPROVED** [review]
@pferraro This all looks fine, except I'm concerned about the 2 undertow remove handler changes.

**Comment by bstansberry** [comment]
Thanks @pferraro
